### PR TITLE
New library form#3

### PR DESCRIPTION
**Pull request by serox2255**
added creation form for library

### CODE REVIEW

File: src/app/library-design/library/creation-form/library-creation.component.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<div class="container" *ngIf="isNewLibraryFormVisible">
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.html:1]: use better naming of class e.g. library-creation-container

File: src/app/library-design/library/creation-form/library-creation.component.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  <div class="form" [formGroup]="formGroup">
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.html:2]: use better naming of class

File: src/app/library-design/library/creation-form/library-creation.component.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <h1 class="header">Library Registration</h1>
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.html:3]: use better naming of class

File: src/app/library-design/library/creation-form/library-creation.component.html (hunk 1 above)
Comment (durixx) [reply src/app/library-design/library/creation-form/library-creation.component.html:1]: also use new lines for elements eg
<div class="container" 
        *ngIf="isNewLibraryFormVisible">

search throught all html and do so

Comment (durixx) [reply src/app/library-design/library/creation-form/library-creation.component.html:1]: div class="container"
     *ngIf="isNewLibraryFormVisible"

File: src/app/library-design/library/creation-form/library-creation.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+  }
+
+
+
+  submit(data) {
+    if (this.formGroup.valid) {
+      console.log('valid');
+      this.http.post('http://localhost:4200/api/library/create', data).subscribe();
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.ts:36]: use request service to store this post call and then use directly this service to save

File: src/app/library-design/library/creation-form/library-creation.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      address: new FormControl('',[Validators.required]),
+      city: new FormControl('',[Validators.required]),
+      country: new FormControl('',[Validators.required])
+    });
+
+  }
+
+
+
+  submit(data) {
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.ts:33]: submit is not good naming

rename method

File: src/app/library-design/library/creation-form/library-creation.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    if (this.formGroup.valid) {
+      console.log('valid');
+      this.http.post('http://localhost:4200/api/library/create', data).subscribe();
+    } else {
+      console.log('not valid');
```
Comment (durixx) [inline src/app/library-design/library/creation-form/library-creation.component.ts:38]: get rid of console.log lines

**Review by durixx — CHANGES_REQUESTED** [review]

**Review by durixx — CHANGES_REQUESTED** [review]
please fix findings

File: src/app/library-design/library/creation-form/library-creation.component.html (hunk 1 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.html:1]: fixed in different branch

File: src/app/library-design/library/creation-form/library-creation.component.html (hunk 2 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.html:2]: fixed in different branch

File: src/app/library-design/library/creation-form/library-creation.component.html (hunk 3 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.html:3]: fixed in different branch

File: src/app/library-design/library/creation-form/library-creation.component.ts (hunk 6 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.ts:38]: fixed

File: src/app/library-design/library/creation-form/library-creation.component.ts (hunk 5 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.ts:33]: fixed

File: src/app/library-design/library/creation-form/library-creation.component.ts (hunk 4 above)
Comment (serox2255) [reply src/app/library-design/library/creation-form/library-creation.component.ts:36]: fixed

**Review by durixx — APPROVED** [review]

**Pull request merged**